docs/userguide/campaigns.md

```diff
@@ -1,25 +1,125 @@
+
+```python
+from baybe.parameters import CategoricalParameter, NumericalDiscreteParameter
+from baybe.searchspace import SearchSpace
+
+grind_size = CategoricalParameter(name="grind_size", values=["coarse", "rather_coarse", "rather_fine", "fine"])
+temp = NumericalDiscreteParameter(name="temp", values = tuple([t/10.0 for t in range (905, 961)]))
+time = NumericalDiscreteParameter(name="time", values = tuple(range(120, 370, 10)))
+
+space = SearchSpace.from_product([grind_size, temp, time])
+```
+
+We refer to the [parameters](./parameters) resp. [search spaces](./searchspace) user
+guides for more details.
+
+To assess the quality of our coffee, we use a subjective evaluation method
+involving tasting and assigning a rating on a scale of 1 to 10, where a higher score
+indicates better quality.
+
+```python
+from baybe.targets import NumericalTarget
+from baybe.objective import Objective
+
+target = NumericalTarget(name="quality", mode="MAX", bounds=(1,10))
+objective = Objective(mode="SINGLE", targets=[target])
+```
+
+Combining the search space and the objective now enables us to create a campaign.
+
+```python
+from baybe.campaign import Campaign
+
+campaign = Campaign(searchspace=space, objective=objective)
+```
+
+## Getting a recommendation and adding a measurement
+
+To obtain a recommendation for the next experiment, we can query the campaign and use
+the [`recommend`](baybe.campaign.Campaign.recommend) function. The function takes only
+one argument, which is the `batch_quantity` keyword. This specifies the desired size of
+the batch of experiments to be conducted.
+
+```python
+rec = campaign.recommend(batch_quantity=3)
+```
+
+The `recommend` function returns a `DataFrame` with `batch_quantity` many rows, each 
+representing a set of parameters from the search space.
+To add measurements, we expand the `DataFrame` by adding a new column for the target.
+We can then provide the campaign with these measurements and receive a new
+recommendation.
+
+```python
+rec["quality"] = [2,4,9]
+campaign.add_measurements(rec)
+new_rec = campaign.recommend(batch_quantity=5)
+```
+
+## Further specifictation of campaign
+
+Although only a search space and an objective are necessary to create a campaign,
+several other aspects can be changed by the user.
+1. **A strategy:**: By default, campaigns use the composite
+[`TwoPhaseStrategy`](baybe.strategies.composite.TwoPhaseStrategy). This strategy uses a
+[`RandomRecommender`](baybe.recommenders.sampling.RandomRecommender) for the initial
+recommendation and then switches to a [`SequentialGreedyRecommender`](baybe.recommenders.bayesian.SequentialGreedyRecommender).
+However, using the `strategy` keyword, you can specify a different strategy.
+For more details on strategies, see the corresponding [`submodule`](baybe.strategies) or
+[user guide](./strategy).
+* **Previously conducted experiments**: If you have previously conducted experiments and
```

Comment:
this is  a catastrophic recommendation and needs to be removed, users should only interact with the campaign via `add_measurements` and `recommend` and the serialization calls, not meddle with `measurements_exp`. As best one could mention that `measurements_exp` stores all uploaded measurements in case the suer needs them again, but they should never be uploaded via that

Reply:
Tagging @AdrianSosic  since I see a bigger issue here: We might need to rework parts of our documentation.
If I were a user looking at the package documentation (which was the role I took when writing this), I see the following for the __init__ function and for the attributes:
<img width="1245" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/7a521541-bc9b-4d96-a391-33d206e14bce">
<img width="965" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/1ac40a86-9ea5-4e07-b3ea-c40b4c1423e5">
**Nothing** is telling me that I should not provide previously measured experiments using that keyword, and if I had no idea about BayBE, I would actually assume that this was the way how to inform BayBE about such experiments. At no point, it is clear how I should inform BayBE about such experiments in any other way.
Also, even if we point stuff like this out in the user guide: There will be people not looking into the user guide who just work with the package documentation. So in my opinion, it is also **not** sufficient to mention stuff like that just in the user guide or examples but need to find some way of having this information in the package documentation.
What we could do is put information about this in the corresponding module docstrings, e.g. "It is intended to only construct a campaign using the provided class methods, and to only interact with it using the functions X, Y and Z. All other attributes and properties are being maintained internally, and manipulating them or using Campaigns in any other way is heavily discouraged"

Reply:
I absolutely see your point! But I don't think this is a documentation issue but rather shows that the current class layout is not yet optimal! Or let me say it differently: if the interfaces were designed perfectly, the documentation would automatically only show the stuff that the user needs to see, and there would be no need to explain anything. 

The explanation why why are not there yet is: 
* The "simple" approach to resolve this would simply be to **not** expose the "internal" stuff via the constructor! 
* The only reason why this is currently done because this was the "easy" way how to enable deserialization of the class.

But I'm convinced there is an elegant way to solve this. Either by adjusting the serialization mechanism, or introducing an additional user layer around. 

My suggestion: let us postpone this discussion by 2-3 weeks until the mypy refactorings have reached the campaign class (that is, if the refactorings of all the used internal objects are completed). That's then the natural point to refine stuff at this level. Would this be acceptable? Until then, we would simply pretend that all internal stuff is not visible when writing the docs, i.e. not mention it in the guides.

Reply:
Ok, so you basically suggest to do all of the userguide stuff after thy my py refactorings in general? Works for me. If @Scienfitz also agrees, I'd put this PR in draft mode until then.

Reply:
No, sorry, should have expressed myself a bit clearer. I think the user guides have high priority at the moment, but cleaning the class less so. Therefore, my suggestion was: just pretend that it is already cleaned up, and the actual cleanup will follow. That is, we only mention in the user guide stuff that is supposed to be public

Reply:
@AdrianSosic and @Scienfitz : To move this on, please verify that only those properties/attributes/functions/... are public that should be public.

Reply:
Summary of my request and reason why you should resolve it: Originally, I put *all* of the properties in the documentation. Upon your request, I changed this and only included the ones that should be visible to the user. Since this change was made due to your comment, you should verify that I did that fix in the way that you intended it. So if you are happy which the selection of attributes that are listed now, please resolve.

Reply:
same as other thread: fine for me after my first round, id not resolve because Adrians review is pending